### PR TITLE
BHBC-2239: Decorate EML

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

https://quartech.atlassian.net/browse/BHBC-2239

## Description of relevant changes

Add `eml-service.ts`
- Add decorate function to modify the JSON EML and add relevant BioHub metadata

Add api-host to queue pipeline
- Is needed because the queue pod runs the decorate EML function when processing a new dwca submission

Updated `jsonpath` library to `jsonpath-plus`

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: database/src/migrations/release.0.7.0/populate_system_metadata_constant.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
-insert into system_metadata_constant (constant_name, character_value, description) values ('ORGANIZATION_URL', 'https://www2.gov.bc.ca/gov/content/governments/organizational-structure/ministries-organizations/ministries/environment-climate-change', 'The organizations public facing URL. This value is used in the production of published ecological metadata language files.');
+insert into system_metadata_constant (constant_name, character_value, description) values ('ORGANIZATION_NAME_FULL', 'Knowledge Management Branch, Ministry of Water, Land and Natural Resources, Government of British Columbia', 'The organizations full name. This value is used in the production of published ecological metadata language files.');
+insert into system_metadata_constant (constant_name, character_value, description) values ('ORGANIZATION_URL', 'https://www2.gov.bc.ca/gov/content/governments/organizational-structure/ministries-organizations/ministries/water-land-and-resource-stewardship', 'The organizations public facing URL. This value is used in the production of published ecological metadata language files.');
```
Comment (NickPhura) [inline database/src/migrations/release.0.7.0/populate_system_metadata_constant.sql:5]: @cgarrettjones FYI, updated as per the ticket. I just updated the existing migration in this case.

File: api/src/services/dwc-service.ts [hunk 2]
```diff
@@ -46,28 +47,31 @@ export class DarwinCoreService extends DBService {
    * @memberof DarwinCoreService
    */
   async intakeJob(jobQueueRecord: ISubmissionJobQueueRecord): Promise<void> {
+    // Step 1: Insert submission metadata record
```
Comment (curtisupshall) [inline api/src/services/dwc-service.ts:50]: +1 for elaborate comments :^)

File: api/src/services/eml-service.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    // possible, if needed.
+    for (const patch of patches) {
+      try {
+        const paths = JSONPath({
+          path: patch.path,
+          json: eml,
+          resultType: 'all'
+        });
+
+        paths.forEach(async (path: any) => {
```
Comment (curtisupshall) [inline api/src/services/eml-service.ts]: Is this callback asynchronous?

**Review by curtisupshall — DISMISSED** [review]
Looks good! 📔

File: api/src/services/eml-service.ts (hunk 3 above)
Comment (NickPhura) [reply api/src/services/eml-service.ts]: Nope! I think I did have an async call at one point and removed it.

Fixed.

**Review by curtisupshall — APPROVED** [review]

**Review by AlfredRosenthal — APPROVED** [review]
🗄️

**Pull request merged**